modules/assimp/editor_scene_importer_assimp.cpp

```diff
@@ -446,21 +446,65 @@ Spatial *EditorSceneImporterAssimp::_generate_scene(const String &p_path, const
 	state.animation_player = NULL;
 
 	real_t scale_factor = 1.0f;
+	Quat rotation_quat;
 	{
 		//handle scale
 		String ext = p_path.get_file().get_extension().to_lower();
 		if (ext == "fbx") {
 			if (scene->mMetaData != NULL) {
-				float factor = 1.0;
+				double factor = 1.0;
 				scene->mMetaData->Get("UnitScaleFactor", factor);
 				scale_factor = factor * 0.01f;
+				int32_t up_axis = 0;
+				int32_t up_axis_sign = 0;
+				int32_t front_axis = 0;
+				int32_t front_axis_sign = 0;
+				int32_t coord_axis = 0;
+				int32_t coord_axis_sign = 0;
+				scene->mMetaData->Get("UpAxis", up_axis);
+				scene->mMetaData->Get("UpAxisSign", up_axis_sign);
+				scene->mMetaData->Get("FrontAxis", front_axis);
+				scene->mMetaData->Get("FrontAxisSign", front_axis_sign);
+				scene->mMetaData->Get("CoordAxis", coord_axis);
+				scene->mMetaData->Get("CoordAxisSign", coord_axis_sign);
+				Vector3 rotation = Vector3(Math::deg2rad(-90.0f), Math::deg2rad(0.0f), Math::deg2rad(0.0f));
+				if (up_axis == 1 && up_axis_sign == 1 &&
+						front_axis == 2 && front_axis_sign == 1 &&
+						coord_axis == 0 && coord_axis_sign == 1) {
```

Comment:
Is the empty code block intended here?

Overall this is pretty hard to read, isn't there any simpler way to do these checks in a human readable way?

Reply:
Blender does it this way... https://github.com/sftd/blender-addons/blob/master/io_scene_fbx/fbx_utils.py#L96

Reply:
I would suggest abstracting these into methods for checking handedness and rotation, which could perhaps be an enum. This could also very likely be optimized since all the contents of the else ifs are the same.